server/src/test/java/com/qubole/quark/serverclient/client/EndToEndTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2015. Qubole Inc
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.qubole.quark.serverclient.client;
+
+import com.qubole.quark.serverclient.server.Main;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.junit.AfterClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.sql.*;
+import java.util.*;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+/**
+ * Created by adeshr on 2/24/16.
+ */
+public abstract class EndToEndTest {
+  protected static final Log LOG = LogFactory.getLog(Main.class);
+
+  public static Main main;
+  public static String h2Url;
+  public static String cubeUrl;
+  public static String viewUrl;
+
+  public static void setupTables(String dbUrl, String filename)
+      throws ClassNotFoundException, SQLException, IOException, URISyntaxException {
+
+    Class.forName("org.h2.Driver");
+    Properties props = new Properties();
+    props.setProperty("user", "sa");
+    props.setProperty("password", "");
+
+    Connection connection = DriverManager.getConnection(dbUrl, props);
+
+    Statement stmt = connection.createStatement();
+    java.net.URL url = EndToEndTest.class.getResource("/" + filename);
+    java.nio.file.Path resPath = java.nio.file.Paths.get(url.toURI());
+    String sql = new String(java.nio.file.Files.readAllBytes(resPath), "UTF8");
+
+    stmt.execute(sql);
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    if (main.getServer() != null) {
+      main.getServer().stop();
+    }
+  }
+
+  @Test
+  public void testClient() throws SQLException, ClassNotFoundException, InterruptedException {
+    Class.forName("com.qubole.quark.jdbc.QuarkDriver");
+    QuarkDriver d = new QuarkDriver();
+    Class.forName("org.h2.Driver");
+    Connection connection = null;
+
+    // Due to threading, server might not be up.
+    for (int i=0; i<2; i++) {
+       try {
+         connection = d.connect(ThinClientUtil.getConnectionUrl("0.0.0.0", 8765), new Properties());
```

Comment:
The test is fishy. You should be using DriverManager to connect. So this part has to be parameterized depending on whether you are running `DbEndToEnd` or `JsonEndToEnd`. 
Example: https://github.com/qubole/quark/blob/master/jdbc/src/test/java/com/qubole/quark/jdbc/test/DbSelectTest.java#L67